packages/model/addon/-private/system/model-for-mixin.ts

```diff
@@ -3,8 +3,8 @@ import { getOwner } from '@ember/application';
 import Model from '../model';
 
 type Store = import('@ember-data/store').default;
-
-/*
+// TODO refactor for native classes/octane??
+/* 
```

Comment:
👍

Reply:
@snewcomer not sure how to convert this example.  Actually, I don't understand the use case...

Maybe 🤷 
```js
import Model, { belongsTo} from '@ember-data/model';
import Mixin from '@ember/object/mixin';

let Comment = new Model ({
  @belongsTo('commentable', { polymorphic: true }) owner;
})

let Commentable = Mixin.create({
  comments: hasMany('comment')
});
```

Reply:
```
import Model, { belongsTo} from '@ember-data/model';
import Mixin from '@ember/object/mixin';

let Comment = class CommentModel extends Model {
  @belongsTo('commentable', { polymorphic: true }) owner;
}

let Commentable = Mixin.create({
  comments: hasMany('comment')
});
```

Reply:
So close and yet so far, this also answers the other example I questioned. Updated